### DISCUSSION TITLE
New cargo behavior idea: percentage

### DISCUSSION THREAD

**Discussion by SecretU4** [OpenTTD/OpenTTD#8965] [Features]
(Sorry for poor English)
At A-Train's most recent game by Nintendo Switch, you can adjust the option of getting in and out of the station in proportion of total amount that the train can carry (like Whole/Half/None).
Can OTTD make this kinds of options? If it can be, it will be very useful when the cargo needs to distribute at several stations by one train.

**Comment by perezdidac** [comment]
Can you elaborate an example? a vehicle stops at a station and loads half of it's capacity and then what?
Why is it useful for when the cargo needs to distribute at several stations by one train? or are you specifically talking about unloading % per station?

**Reply by nielsmh** [reply]
I assume this is for situations like FIRS "Supplies" cargo types, where you want a little distributed to each of a large number of consumers.

**Reply by SecretU4** [reply]
It can be useful like this situation:  
A factory st. supplies goods 10t per hour
B factory st. supplies goods 30t per hour  
If the route shape like A-B-city(accepts cargo) and its round trip takes 2 hours, I can make a efficient cargo route like below.  

1. (the train can take 80t of goods.)
2. collect cargo at A until the train takes 20t of goods.(quarter of capacity)(or 25%)
3. collect cargo at B until the train is full.
4. deilver to city
5. go back to 2.

**Comment by LC-Zorg** [comment]
This would be a very useful feature especially for the FIRS and similar industries sets. When playing on FIRS servers, players often asked if only part of the load can be unloaded. Loading to a certain level can also be very useful. For example, when only vehicles with big capacities are available and the enterprise has little production. Currently, it can also be done for loading with conditional orders (jump to the previous order if the loading is less than e.g. 20%), but it doesn't work well, it can be troublesome and determining them is very busy.

Percentage of loading with the use of a conditional order
![Precentage load by conditional orders](https://user-images.githubusercontent.com/67662046/114322704-ef8aa900-9b21-11eb-89a1-1d9cfc5949e1.png)

These types of orders were one of the motivations for creating an alternative orders window, where their selection would be done by clicking on the icon. This is simpler and faster to use, and has more options than selecting by text list.
![Orders window v2 8 2 example](https://user-images.githubusercontent.com/67662046/114322812-7b043a00-9b22-11eb-88f2-d60cf9510d5d.png)

In this example, you can specify the loading or unloading percentage. Commands are similar to conditional orders:
- less than / < / x%
- less or equal to / ≤ / x%
- equal to / = / x%
- more or equal to / ≥ / x%
- more than / > / %

I don't know if this window and the method of selecting orders is good for others (this draft is still not perfect). I mention this to highlight some possibilities with such orders. This is also possible in the current text-based orders form.

![Precentage load and unload feature](https://user-images.githubusercontent.com/67662046/114322891-e9e19300-9b22-11eb-9221-93367a86faeb.png)
